numpy/core/code_generators/generate_umath.py

```diff
@@ -1027,14 +1044,24 @@ def make_arrays(funcdict):
                     for vt in t.simd:
                         code2list.append(textwrap.dedent("""\
                         #ifdef HAVE_ATTRIBUTE_TARGET_{ISA}
-                        if (npy_cpu_supports("{isa}")) {{
+                        if (NPY_CPU_HAVE({ISA})) {{
                             {fname}_functions[{idx}] = {type}_{fname}_{isa};
                         }}
                         #endif
                         """).format(
                             ISA=vt.upper(), isa=vt,
                             fname=name, type=tname, idx=k
                         ))
+                if t.dispatch is not None:
+                    for dt in t.dispatch:
+                        code2list.append(textwrap.dedent("""\
+                        #ifndef NPY_DISABLE_OPTIMIZATION
+                        #include "{dname}.dispatch.h"
+                        #endif
+                        NPY_CPU_DISPATCH_ASSIGN_XB({fname}_functions[{idx}], {type}_{fname})
+                        """).format(
+                            dname=dt, fname=name, type=tname, idx=k
+                        ))
```

Comment:
maybe start with `for dname in t.dispatch`, and leave `name`, `tname` as-is. Then the formatting does not need to do so much renaming